src/Raven.Server/Documents/Handlers/Processors/OngoingTasks/AbstractOngoingTasksHandlerProcessorForAddEtl.cs

```diff
@@ -0,0 +1,81 @@
+﻿using System;
+using System.Threading.Tasks;
+using JetBrains.Annotations;
+using Raven.Client.Documents.Operations.ConnectionStrings;
+using Raven.Client.Documents.Operations.ETL;
+using Raven.Server.Documents.Handlers.Processors.Databases;
+using Raven.Server.ServerWide.Context;
+using Sparrow.Json;
+using Sparrow.Json.Parsing;
+
+namespace Raven.Server.Documents.Handlers.Processors.OngoingTasks
+{
+    internal abstract class AbstractOngoingTasksHandlerProcessorForAddEtl<TRequestHandler, TOperationContext> : AbstractHandlerProcessorForUpdateDatabaseConfiguration<BlittableJsonReaderObject, TRequestHandler, TOperationContext>
+        where TOperationContext : JsonOperationContext
+        where TRequestHandler : AbstractDatabaseRequestHandler<TOperationContext>
+    {
+        protected AbstractOngoingTasksHandlerProcessorForAddEtl([NotNull] TRequestHandler requestHandler)
+            : base(requestHandler)
+        {
+        }
+
+        protected override void OnBeforeResponseWrite(TransactionOperationContext _, DynamicJsonValue responseJson, BlittableJsonReaderObject configuration, long index)
+        {
+            responseJson[nameof(EtlConfiguration<ConnectionString>.TaskId)] = index;
+        }
+
+        protected override void OnBeforeUpdateConfiguration(ref BlittableJsonReaderObject configuration, JsonOperationContext context)
+        {
+            AssertCanAddOrUpdateEtl(ref configuration);
+        }
+
+        protected override async ValueTask OnAfterUpdateConfiguration(TransactionOperationContext _, BlittableJsonReaderObject configuration, string raftRequestId)
+        {
+            // Reset scripts if needed
+            var scriptsToReset = HttpContext.Request.Query["reset"];
```

Comment:
use dedicated method from RequestHandler. IIRC it was GetStringValuesFromQueryString

Reply:
done